rfc-119-merge-mainstream-browse-topics-and-taxons.md

```diff
@@ -0,0 +1,169 @@
+
+## Proposal
+
+We propose merging the attributes of mainstream browse and specialist topic content items into **existing** taxon content items with the same `title`. (For clarity we aren't proposing the creation of new taxons or taxon tree.)
+
+Generally there are two main type of information we want to tranfer from mainstream browse and specialist topic to the equivalent taxon. The first is the links to content items. Currently content items (guidance pages, manuals etc..) do not contain links to mainstream browse or specialist topics. It's the reverse, mainstream browse pages and specialist topics have a list of content that's associated to them. This list we can migrate over to the taxon.
+
+Secondly, we'd carry across the relationships (links) between mainstream browse pages/specialist topics. This manifests as multiple hierarchies within the same taxons. This is because different relationships are useful for different things ie. the currently hierarchy for taxonomy is good for emails subscriptions and organising taxons, but not so great at modelling navigation. So we'd bring over the mainstream browse hierarchy to serve that purpose.
+
+#### Top level attributes
+
+All top level attributes will continue to have the existing values in the taxon:
+
+- `title`
+- `description`
+- `analytics_identifier`
+- `content_id`
+- `document_type`
+- `first_published_at`
+- `locale`
+- `phase`
+- `public_updated_at`
+- `publishing_scheduled_at`
+- `rendering_app`
+- `scheduled_publishing_delay_seconds`
+- `schema_name`
+- `updated_at`
+- `withdrawn_notice`
+- `publishing_request_id`
+
+This is because they don't cause conflict or the mainstream or specialist topic value becomes redundant.
+
+- `publishing_app`
+
+Currently references Collections Publisher for mainstream and specialist topics, however references Content Tagger for taxons. We propose that `publishing-app` remains Content Tagger as it still has the responsibility of creating and removing taxons, even though some sub properties (i.e. links and details) may be set by Collections Publisher. In the future, we could also look at consolidating the publishing functionality of Collections Publisher and Content Tagger.
+
+- `base_path`
+
+Mainstream browse and specialist topic pages have different `base_path` to taxons. We propose that taxons keep their existing `base_path`. We could use new mainstream browse/specialist topic links within taxons to generate old base paths or find the appropriate taxon from an existing mainstream browse or specialist topic base path. Alternatively, we keep the old base paths as renamed attributes (e.g. `mainstream_browse_base_path`).
```

Comment:
Are we proposing to redirect all the old base paths, or have different "views" of the taxon for different paths? Content items already support [multiple "routes"](https://github.com/alphagov/govuk-content-schemas/blob/b012c9c8c28781393f25f818a30dfd36a2cc4ae3/formats/shared/definitions/routes_redirects.jsonnet#L3) - perhaps we could use that functionality instead.

Reply:
So _I think_ this talking about application code that makes use of the `base_path` attribute in a content item. E.g.`content_item["base_path"]`

I think can easily derive the relevant base paths by prepended or dropping the "topic" or "browse" bit.

i.e.
Mainstream browse/topic base path <=> Taxon base path
/browse/foo/bar                 <=> /foo/bar
/topic/foo/bar                     <=> /foo/bar

```suggestion
Mainstream browse and specialist topic pages have a different `base_path` to taxons. We propose that taxons keep their existing `base_path`. 

Applications can easily derive the old base paths for mainstream browse or specialist topic by prepending "browse" or "topic" respectively e.g. `"/browse/#{taxon_base_path}"` or `"/topic/#{taxon_base_path}"`. Alternatively, we keep the old base paths as renamed attributes (e.g. `mainstream_browse_base_path`).
```


But yup, I think we need to expand on what happens to the old base paths when we access the URL e.g GET `www.gov.uk/browse/something` or `www.gov.uk/api/content/browse/something`. 

We might need to add two content items to represent the [/browse](https://www.gov.uk/browse) and [/topic](https://www.gov.uk/topic) pages. They'd need a prefix routing rule for `/browse*` or `/topic*`, so that requests are redirected to the frontend app (which can fetch the correct taxon instead of the mainstream browse or topic content item). This would prevent us adding redirects from mainstream browse/topics pages to taxons (e.g. `/browse/education` => `/education`) but don't think it's an issue.

Reply:
I'm still confused, sorry. The point about `/browse` and `/topic` is definitely worth adding, though, as that's an unexpected edge case: `/browse` doesn't have a content item, and `/topic` is the root content item.

I think we generally try and avoid prefix routes where possible, so the app doesn't have to handle 404s itself. We should be able to use sub-routes on the taxon content item to keep all the old paths in place.

But it might be worth pausing this thread, as my confusion seems to stem from not fully understanding the background / problem, which we're discussing in [another thread](https://github.com/alphagov/govuk-rfcs/pull/119#discussion_r524295884).

Reply:
It'd be good to be clearer on the `browse/*` `topic/*` prefix routes as a consequence. I think I had this explained to me last time this RFC went around but I didn't pick up on it reading through this again until I read this thread.

I'd interpreted it as we'd have say `www.gov.uk/{taxon-base-path}` and thought that might render as a specialist topic, a taxon or a mainstream browse page - this made me a bit concerned about overloading of concepts.

I'd also suggest staying away from a second base_path in the details - that seems a smell that this is two distinct things with a relationship (a browse item coupled to a topic taxon) than the same thing in a different scenario.